packaging/version.py

```diff
@@ -198,7 +198,7 @@ class Version(_BaseVersion):
 
     def __init__(self, version):
         # Validate the version and parse it into pieces
-        match = self._regex.search(version)
+        match = self._regex.search(str(version))
```

Comment:
Is this required? I would kind of prefer not to ``str()`` everything passed into version since that may mask things like doing ``Version(1)`` that should still fail.